apis/workstations/v1alpha1/config_identity.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/apis/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkstationConfigIdentity defines the resource reference to WorkstationConfig, which "External" field
+// holds the GCP identifier for the KRM object.
+type WorkstationConfigIdentity struct {
+	parent *WorkstationConfigParent
+	id     string
+}
+
+func (i *WorkstationConfigIdentity) String() string {
+	return i.parent.String() + "/workstationConfigs/" + i.id
+}
+
+func (i *WorkstationConfigIdentity) ID() string {
+	return i.id
+}
+
+func (i *WorkstationConfigIdentity) Parent() *WorkstationConfigParent {
+	return i.parent
+}
+
+type WorkstationConfigParent struct {
```

Comment:
Some thoughts: This can use the "WorkstationClusterIdentity", so shows the direct relationship between a child resource identity and its parent resource identity, especially if you want to only define the project/location in parent.